definitions/subworkflows/input_to_trimmed_fastq_and_hisat_alignments.cwl

```diff
@@ -39,10 +41,21 @@ outputs:
         type: File
         outputSource: hisat2_align/aligned_bam
 steps:
+    input_to_bam:
+        run: ../tools/input_to_bam.cwl
+        in:
+            input: input
+            cram_reference: cram_reference
+        out: [bam_file]
+    revert_bam:
+        run: ../tools/revert_input.cwl
```

Comment:
Could `revert_input.cwl` be run on CRAMs directly?  If so seems like `input_to_bam` could be removed.  If not, seems like it should be named `revert_bam.cwl` instead of `revert_input.cwl` to denote its specificity.

(Is this operation time and space-efficient enough that we're willing to do it on all inputs instead of only those that need it?)

Reply:
The problem is the specificity, the tool in revert_input.cwl can be run on anything but you would have to use the input variable, currently bam, to grab the appropriate out file , basically would require some javascript i expect which i try to avoid in cwl. I think renaming to revert_bam.cwl is preferable from my perspective.

The flexibility in the PR does come at a computational cost for sure. I think we should run revert_input.cwl either way though so the cost is just converting whatever input to bam